src/alchemlyb/parsing/lammps.py

```diff
@@ -0,0 +1,1303 @@
+
+    # Check for MBAR content
+    missing_combinations_mbar = []
+    for lambda_value, lambda_array in lambda_dict.items():
+        missing_combinations_mbar.extend(
+            [(lambda_value, x) for x in lambda_values if x not in lambda_array]
+        )
+
+    if missing_combinations_mbar:
+        warnings.warn(
+            "The following combinations of lambda and lambda prime are missing for MBAR analysis: {}".format(
+                missing_combinations_mbar
+            )
+        )
+    else:
+        return lambda_values, lambda_pairs, lambda2
+
+    # Check for BAR content
+    missing_combinations_bar = []
+    lambda_values.sort()
+    for ind, (lambda_value, lambda_array) in enumerate(lambda_dict.items()):
+        if ind == 0:
+            tmp_array = [lambda_values[ind], lambda_values[ind + 1]]
+        elif ind == len(lambda_dict) - 1:
+            tmp_array = [lambda_values[ind - 1], lambda_values[ind]]
+        else:
+            tmp_array = [
+                lambda_values[ind - 1],
+                lambda_values[ind],
+                lambda_values[ind + 1],
+            ]
+
+        missing_combinations_bar.extend(
+            [(lambda_value, x) for x in tmp_array if x not in lambda_array]
+        )
+
+    if missing_combinations_bar and not force:
+        raise ValueError(
+            "BAR calculation cannot be performed without the following lambda-lambda prime combinations: {}".format(
+                missing_combinations_bar
+            )
+        )
+
+    return lambda_values, lambda_pairs, lambda2
+
+
+@_init_attrs
+def extract_u_nk_from_u_n(
+    fep_files,
+    T,
+    column_lambda,
+    column_U,
+    column_U_cross,
+    dependence=lambda x: (x),
+    index=-1,
+    units="real",
+    prec=4,
+    ensemble="nvt",
+    pressure=None,
+    column_volume=4,
+):
+    """Produce u_nk from files containing u_n given a separable dependence on lambda.
+
+    Parameters
+    ----------
+    fep_files : str or list
+        If not a list, a str representing the path to fepout file(s) to extract data from. Filenames and paths are
+        aggregated using `glob <https://docs.python.org/3/library/glob.html>`_. For example, "/path/to/files/something_*.txt".
+    T : float
+        Temperature in Kelvin at which the simulation was sampled.
+    columns_lambda : int
+        Indices for columns (file column number minus one) representing the lambda at which the system is equilibrated
+    column_U : int
+        Index for the column (file column number minus one) representing the potential energy of the system.
+    column_U_cross : int
+        Index for the column (file column number minus one) representing the potential energy of the cross interactions
+        between the solute and solvent.
+    dependence : func, default=`lambda x : (x)`
+        Dependence of changing variable on the potential energy, which must be separable.
```

Comment:
Are these all going to be different case by case or they are fixed?

Reply:
They can vary because the LAMMPS dump is arbitrarily user defined. I've added a reference in the doctorings to a python package I wrote to standardize this.